src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jobs/oneoff/migration/ChangelogMigrationProcessor.kt

```diff
@@ -0,0 +1,83 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jobs.oneoff.migration
+
+import mu.KLogging
+import org.springframework.batch.item.ItemProcessor
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.ReferralAmendmentDetails
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Changelog
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ReferralDetails
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AuthUserRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ChangelogRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralDetailsRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.AmendTopic
+import java.util.UUID
+
+@Component
+class ChangelogMigrationProcessor(
+  private val changelogRepository: ChangelogRepository,
+  private val referralDetailsRepository: ReferralDetailsRepository,
+  private val authUserRepository: AuthUserRepository
+) : ItemProcessor<ReferralDetails, Changelog?> {
+  companion object : KLogging()
+
+  override fun process(referralDetails: ReferralDetails): Changelog? {
+    logger.info("processing referral details {} for referral id {}", referralDetails.id, referralDetails.referralId)
+    // check if any changlog is null
+    if (referralDetails.supersededById != null) {
```

Comment:
We can roll this into the original query to reduce the number of items processed (and could make it more evident with `supersededDetails: ReferralDetails` variable)

`.queryString("SELECT r FROM ReferralDetails r WHERE r.supersededById is not null")`?

Reply:
changed the query to **`select rd from ReferralDetails rd where rd.supersededById is not null order by createdAt ASC`**